### DISCUSSION TITLE
Some issues with the drawio collaboration module code

### DISCUSSION THREAD

**Discussion by miki-hmt** [jgraph/drawio#3827] [General]
### Preflight Checklist
Hello author, I noticed that there are two websockets in the real-time collaboration module of drawio, one interacting with the pusher and the other interacting with drawio's own rt. Is there any connection between these two websockets and onedrive? If it's convenient, could you give me some explanation? I'm planning to implement the logic for this myself

**Comment by jprtf** [comment]
Tagging @VendettaMike as he is into this topic.

**Comment by miki-hmt** [comment]
Hello boss, I recently implemented the cursor trajectory and icon content change transfer function for localized multi person collaboration in drawio through websocket and pusher myself. However, as you can see, although the content has changed, the client's changed components quickly returned to their original state. I found that it was caused by an event that did not call automatic save, and then I observed on the official app. diagrams.net, The pusher end will send a {"event": "changed", "data": "" U2FsdGVkX1+rr97DNKwRln/FJbGbNfozfR5yEBHcKTcMif/6ij/piiYf9jGsvHoOlybTjfC99wtr/bjYDbJiZDCXd8yyZP8H9HFkHt7sjSHWHyB/k6f4l/tGKu4ycfPJdZUiOefvsWJflXLDaRYT8Z CE56FvjLbqeIiNLV0   u003d, I developed it myself and did not send this message to various pusher clients. I guess this is probably the reason. So, if it's convenient here, could you please ask if the pusher message was pushed by the RT service to the pusher server and then monitored by the pusher client?

Here is my operation video：
https://github.com/jgraph/drawio/assets/45377370/f19d264b-33f3-4eb0-a6f1-7f53d8eacb52
